processing/src/main/java/org/apache/druid/math/expr/vector/VectorProcessors.java

```diff
@@ -778,6 +842,15 @@ public ExprEvalVector<long[]> asEval()
     );
   }
 
+  /**
+   * Creates an {@link ExprVectorProcessor} for the logical 'and' operator, which produces a long typed vector output
+   * with values set by the following rules:
+   *    true/true -> true (1)
+   *    true/null, null/true, null/null -> null
+   *    false/null, null/false -> false (0)
```

Comment:
Why does `null` null out a `true`, but not `null` out a `false`!?!?!?!?!?  Is that some SQL-ism to handling nulls?

Reply:
yeah, SQL compat stuff. `null` is basically a stand-in for "unknown" for logical operations,```true && ???``` could be `true` or `false` depending on the value of `???` so the result is `???`. But ```false && ???``` is definitely false no matter the value of `???`.

Reply:
Ahhh, I can see the logic there...  it's providing the best answer based on what it can know from the boolean operator.  It would be a lot easier to reason about if it consistently nulled things out, but if this is how SQL is expected to work, so be it.